pkg/operator/ceph/cluster/rbd/mirror.go

```diff
@@ -76,11 +72,20 @@ func (r *ReconcileCephRBDMirror) start(cephRBDMirror *cephv1.CephRBDMirror) erro
 		return nil
 	}
 
+	secretResourceVersion, err := r.generateKeyring(r.clusterInfo, daemonConf)
+	if err != nil {
+		return errors.Wrapf(err, "failed to generate keyring for %q", resourceName)
+	}
+
 	// Start the deployment
 	d, err := r.makeDeployment(daemonConf, cephRBDMirror)
 	if err != nil {
 		return errors.Wrap(err, "failed to create rbd-mirror deployment")
 	}
+	cephRBDMirror.Status.Cephx.Daemon = keyring.UpdatedCephxStatus(true, r.cephClusterSpec.Security.CephX.Daemon, r.clusterInfo.CephVersion, cephRBDMirror.Status.Cephx.Daemon)
```

Comment:
This is separated from the actual status update, which makes it hard to understand what this line of code affects. Move this to before the status is updated

Reply:
done.